pkg/api/types.go

```diff
@@ -1391,11 +1391,9 @@ const (
 	// Enforced by the scheduler and Kubelet.
 	// TaintEffectNoScheduleNoAdmit TaintEffect = "NoScheduleNoAdmit"
 	// NOT YET IMPLEMENTED. TODO: Uncomment field once it is implemented.
-	// Do not allow new pods to schedule onto the node unless they tolerate the taint,
-	// do not allow pods to start on Kubelet unless they tolerate the taint,
-	// and evict any already-running pods that do not tolerate the taint.
-	// Enforced by the scheduler and Kubelet.
-	// TaintEffectNoScheduleNoAdmitNoExecute = "NoScheduleNoAdmitNoExecute"
+	// Evict any already-running pods that do not tolerate the taint.
+	// Enforced by Kubelet.
+	TaintEffectNoExecute TaintEffect = "NoExecute"
```

Comment:
Way back when I was discussing these names with @bgrant0607 he was suggesting (and I think it's reasonable) that they should all have the same form, i.e. if we are doing NoSchedule and NoAdmit, then this one should also be NoXXX. "Execute" was the closest opposite of "evict" that we could think of. I agree it is not very good, but haven't heard a better suggestion yet.